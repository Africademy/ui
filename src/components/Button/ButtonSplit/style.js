import { css } from 'styled-components';
import {
  blueLighter, gray,
} from '../../style/colors';

export const split = css`
  padding-left: 32px;
  :before {
    background-color: ${props => (props.primary ? blueLighter : gray)};
    content: "";
    height: 24px;
    position: absolute;
    right: 38px;
    top: 50%;
    transform: translateY(-50%);
    width: 1px;
  }
`;
