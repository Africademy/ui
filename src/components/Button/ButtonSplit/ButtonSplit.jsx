import React from 'react';
import PropTypes from 'prop-types';
import style from 'styled-components';
import * as Styles from './style';

const ButtonArrow = style.div`
  ${Styles.split};
   ${props => Styles[props.type]};
`;

const ButtonSplit = ({ onClick, type }) => (
  <ButtonArrow onClick={onClick} type={type}>â–¾</ButtonArrow>
);

ButtonSplit.propTypes = {
  onClick: PropTypes.func.isRequired,

  /** Type of button */
  type: PropTypes.oneOf(['link', 'primarySplit', 'primary', 'secondary', 'text']).isRequired,
};

export default ButtonSplit;
