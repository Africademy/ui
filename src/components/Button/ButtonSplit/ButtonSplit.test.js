import snap from 'jest-auto-snapshots';
import ButtonSplit from './ButtonSplit';

snap(ButtonSplit, './ButtonSplit.jsx');
