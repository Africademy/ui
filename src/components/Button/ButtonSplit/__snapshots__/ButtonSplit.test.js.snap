// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest-auto-snapshots > ButtonSplit Matches snapshot when passed only required props 1`] = `
<ForwardRef
  onClick={[MockFunction]}
  type="link"
>
  â–¾
</ForwardRef>
`;
